src/read.rs

```diff
@@ -349,7 +349,7 @@ impl<R: Read + io::Seek> ZipArchive<R> {
     }
 
     /// Search for a file entry by name
-    pub fn by_name<'a>(&'a mut self, name: &str) -> ZipResult<ZipFile<'a>> {
+    pub fn by_name(&mut self, name: &str) -> ZipResult<ZipFile> {
```

Comment:
This will unify the lifetimes of `self` and `name`, changing the API. Did clippy recommend this change?